modules/ob-build-validation.adoc

```diff
@@ -0,0 +1,44 @@
+// Ths module is included in the following assembly:
+//
+// * builds/overview-openshift-builds.adoc
+
+:_content-type: REFERENCE
+[id="build-validations_{context}"]
+= Build validations
+
+To avoid triggering `BuildRun` resources that will fail because of incorrect or missing dependencies or configuration settings, the build controller validates them in advance. If all validations are successful, you view a `status.reason` field named `Succeeded`. However, if any validations fail, you must check the `status.reason` and `status.message` fields to understand the root cause.
+
+.Validation of builds by build controller
+[options="header"]
+|===
+
+| Status.Reason field | Description
+
+| `BuildStrategyNotFound` | The referenced strategy at namespace level does not exist.
+
+| `ClusterBuildStrategyNotFound` | The referenced strategy at cluster level that does not exist.
+
+| `SetOwnerReferenceFailed` | Setting owner references between a `Build` and a `BuildRun` resources failed. This status is triggered when you use the `build.shipwright.io/build-run-deletion` annotation in a build.
+
+| `SpecSourceSecretRefNotFound` | The secret used to authenticate to Git does not exist.
+
+| `SpecOutputSecretRefNotFound` | The secret used to authenticate to the container registry does not exist.
+
+| `SpecBuilderSecretRefNotFound` | The secret used to authenticate to the container registry does not exist.
+
+| `MultipleSecretRefNotFound` | Multiple secrets used for authentication are missing. 
+
+| `RestrictedParametersInUse` | One or many defined `params` are colliding with any reserved parameters. 
+
+| `UndefinedParameter` | The parameters are not defined in the referenced strategy. You must define those parameters in the `spec.parameters` specification in your strategy.
+
+| `RemoteRepositoryUnreachable` | The defined `spec.source.url` specification was not found. This validation only takes place for HTTP or HTTPS protocol.
```

Comment:
```suggestion
| `RemoteRepositoryUnreachable` | The defined `spec.source.url` specification was not found. This validation only takes place for HTTP or HTTPS protocols.
```

Reply:
@adellape why you suggested to have `protocol` in plural? We have used `or` between HTTP and HTTPS, so the condition is based on either of the two. Either HTTP or HTTPS protocol.